iiblib/iibclient.py

```diff
@@ -517,19 +549,31 @@ def remove_operators(
               if raw == True return dict with json response otherwise
               return IIBBuildDetailsModel instance.
         """
+        post_data = {
+            "from_index": index_image,
+            "binary_image": binary_image,
+            "operators": operators,
+            "add_arches": arches,
+        }
 
-        resp = self.iib_session.post(
-            "builds/rm",
-            json={
-                "from_index": index_image,
-                "binary_image": binary_image,
-                "operators": operators,
-                "add_arches": arches,
-                "overwrite_from_index": overwrite_from_index,
-            },
-        )
-
+        if overwrite_from_index:
+            if overwrite_from_index_token:
+                post_data["overwrite_from_index"] = overwrite_from_index
+                post_data["overwrite_from_index_token"] = overwrite_from_index_token
+            else:
+                raise ValueError(
```

Comment:
I wonder if we want to do these kinds of tests on the client side. This library supposed to be thin layer between client and server, I think I would leave input validation on the server

Reply:
These checks in IIB should happen before anything is started so it can fail asap. I think lesser evil is wasting bit of machine resources than wasting people resources as once you started doing this you'll have to maintain it

Reply:
Also, I forgot to mention - you'll have to do these checks in IIB anyway as iiblib is not exclusive library for communication with IIB

Reply:
Lol, I think if Pub talks to IIB using iiblib, it might be nice to have the check here so that we can save Pub and IIB resources. Maintaining an if-else block shouldn't be a problem IMHO

fwiw, these checks are already present in IIB.